CMakeLists.txt

```diff
@@ -26,6 +27,49 @@ option(USE_NCCL "Use NCCL library" ON)
 option(USE_SENTENCEPIECE "Download and compile SentencePiece" ON)
 option(USE_STATIC_LIBS "Link statically against non-system libs" OFF)
 option(GENERATE_MARIAN_INSTALL_TARGETS "Generate Marian install targets (requires CMake 3.12+)" OFF)
+option(M32_BINARIES "Generate 32bit binaries even when building outside outside of WASM. Useful for" OFF)
+option(COMPILE_WASM "Compile customized marian for WASM target" OFF)
+
+# cmake options that are dependent on COMPILE_WASM cmake option
+CMAKE_DEPENDENT_OPTION(COMPILE_DECODER_ONLY "Compile marian-decoder only" ON
+                       "COMPILE_WASM" OFF)
+CMAKE_DEPENDENT_OPTION(COMPILE_WITH_PTHREADS "Compile with pthreads support" OFF
```

Comment:
To the extent that there is a style with marian-dev's options, `COMPILE_` controls whether a build product exists like `COMPILE_SERVER` while `USE_` controls whether a particular functionality/library is supported like `USE_SENTENCEPIECE`.  This feels more like USE_THREADS.

Reply:
Fixing it in the next commit.

Reply:
Fixed.